src/sonic-framework/rebootbackend/rebootbe.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+#include "dbconnector.h"
+#include "notificationconsumer.h"
+#include "notificationproducer.h"
+#include "reboot_common.h"
+#include "reboot_interfaces.h"
+#include "reboot_thread.h"
+#include "selectableevent.h"
+#include "status_code_util.h"
+
+namespace rebootbackend {
+
+#define REBOOT_REQUEST_NOTIFICATION_CHANNEL "Reboot_Request_Channel"
+#define REBOOT_RESPONSE_NOTIFICATION_CHANNEL "Reboot_Response_Channel"
+#define REBOOT_KEY "Reboot"
+#define REBOOT_STATUS_KEY "RebootStatus"
+#define CANCEL_REBOOT_KEY "CancelReboot"
+#define DATA_TUPLE_KEY "MESSAGE"
+
+class RebootBE {
+ public:
+  struct NotificationRequest {
+    std::string op;
+    std::string ret_string;
+  };
+
+  enum class RebManagerStatus {
+    WARM_INIT_WAIT,
+    IDLE,
+    COLD_REBOOT_IN_PROGRESS,
+    WARM_REBOOT_IN_PROGRESS
+  };
+
+  RebootBE(DbusInterface &interface);
+
+  RebManagerStatus GetCurrentStatus();
+
+  void Start();
+  void Stop();
+
+ private:
+  std::mutex m_status_mutex;
+  RebManagerStatus m_current_status = RebManagerStatus::IDLE;
+  swss::SelectableEvent m_done;
+
+  swss::DBConnector m_db;
+  swss::NotificationProducer m_rebootResponse;
+  swss::NotificationConsumer m_notificationConsumer;
```

Comment:
We should be consistent about snake_case and camelCase.

Reply:
Changed to camelCase